sugar/errors.go

```diff
@@ -5,10 +5,34 @@ import "regexp"
 var re = regexp.MustCompile("\\s+at\\s+`[^`]+`")
 
 func removeStackRecords(s string) string {
+	// some error constructors (such as fmt.Errorf, grpcStatus.Error) serialized error string at construct time
+	// thats why true way with cast wrapped error to *xerrors.stackError has no effect
```

Comment:
Some error constructors (such as `fmt.Errorf`, `grpcStatus.Error`) are serializing the error string at the construction time. Thats why the "true way" with casting the wrapped error into `*xerrors.stackError` has no effect